### PR TITLE
Added test scenario: check missing in report

### PR DESCRIPTION
**Pull request by tisutisu**
This PR will cover story: https://issues.redhat.com/browse/HELM-286

### CODE REVIEW

File: tests/functional/utils/chart_certification.py [hunk 1]
```diff
@@ -502,6 +502,22 @@ def process_report(self, update_chart_sha=False, update_url=False, url=None):
...
                         pytest.fail("Failed to update report yaml with chart-uri")
+            
+            #For removing the check for missing check scenario
+            if missing_check:
+                logging.info(f"Updating report with {missing_check}")
+                with open(report_path, 'r+') as fd:
+                    report_content = yaml.safe_load(fd)
+                    results = report_content["results"]
+                    new_results = []
+                    for result in results:
```
Comment (zonggen) [inline tests/functional/utils/chart_certification.py:513]: Seems like a good fit for `filter`: https://docs.python.org/3/library/functions.html#filter, though this is more readible so I don't have strong opinion.

**Review by mmulholla — APPROVED** [review]
LGTM

File: tests/functional/utils/chart_certification.py (hunk 1 above)
Comment (tisutisu) [reply tests/functional/utils/chart_certification.py]: Good suggestion. I think applying filter for this part of the code will be good, added the change.

File: tests/functional/features/report_with_missing_checks.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        And the report has a <check> missing
+        When the user sends a pull request with the report
+        Then the pull request is not merged
+        And user gets the <message> in the pull request comment
+
+        Examples:
+            | vendor_type  | vendor    | check                          | message                                            |
+	    | partners     | hashicorp | v1.0/helm-lint                 | Missing mandatory check : v1.0/helm-lint           |
+	    # Commented this scenario, since it is failing , raised bug : https://issues.redhat.com/browse/HELM-289 , we can uncomment again when the issue fixed 
+	    #| redhat       | redhat    | v1.0/helm-lint                 | Missing mandatory check : v1.0/helm-lint           |
```
Comment (mmulholla) [inline tests/functional/features/report_with_missing_checks.feature:20]: This test is actually working, based on current behavior auto-merge is expected. That behavior is up for discussion now. Fine to merge this in the meantime.

**Review by mmulholla — APPROVED** [review]

**Pull request merged**